packages/edit-site/src/components/post-list/index.js

```diff
@@ -112,16 +114,15 @@ function useView( postType ) {
 
 	const setViewWithUrlUpdate = useCallback(
 		( newView ) => {
-			const { params } = history.getLocationWithParams();
-
-			if ( newView.type === LAYOUT_LIST && ! params?.layout ) {
+			if ( newView.type === LAYOUT_LIST && ! layout ) {
 				// Skip updating the layout URL param if
 				// it is not present and the newView.type is LAYOUT_LIST.
-			} else if ( newView.type !== params?.layout ) {
-				history.push( {
-					...params,
-					layout: newView.type,
-				} );
+			} else if ( newView.type !== layout ) {
+				history.navigate(
+					addQueryArgs( path, {
+						layout: newView.type,
+					} )
+				);
```

Comment:
It would be nice to keep the `getLocationWithParams` pattern. The idea is that the click handler can read the current `path` at the time when it's called, non-reactively. The component doesn't need to be rerendered on every `path` change and the callback doesn't need to be recomputed.

Reply:
The problem is that we need to know the registered routes, it's not just about reading the arguments from the url, so IMO, it's better to remove it entirely.